components/webdriver_server/lib.rs

```diff
@@ -122,8 +122,25 @@ fn cookie_msg_to_cookie(cookie: cookie::Cookie) -> Cookie {
     }
 }
 
-pub fn start_server(port: u16, constellation_chan: Sender<EmbedderToConstellationMessage>) {
-    let handler = Handler::new(constellation_chan);
+pub fn create_embedder_channel(
+    event_loop_waker: Box<dyn EventLoopWaker>,
+) -> (EmbedderProxy, Receiver<EmbedderMsg>) {
+    let (sender, receiver) = unbounded();
+    (
+        EmbedderProxy {
+            sender,
+            event_loop_waker,
+        },
+        receiver,
+    )
+}
+
+pub fn start_server(
+    port: u16,
+    constellation_chan: Sender<EmbedderToConstellationMessage>,
```

Comment:
Let's have this channel only accept `WebDriverCommandMsg` instead of adding a new variant to `EmbedderMsg`. `EmbedderMsg` should be used for messages that come from the "inside" of Servo out to the embedder. Only sending `WebDriverCommandMsg` on this channel should simplify the code.

```suggestion
    constellation_chan: Sender<EmbedderToConstellationMessage>,
```

This will ensure that various kinds of weird situations are automatically impossible (such as receiving other kinds of `EmbedderMsg`s on this channel).

Reply:
This channel exists a long time ago, and provided by `constellation`. We can not change it because all components use this to send msg to `constellation`.
Fortunately, we can remove this channel from `webdriver` when all msg are sent to `embedder`.
Maybe you confuse this one with `embedder_sender`?

Reply:
As a trick, we can name it `constellation_chan_deprecated` to indicate that this is something we want to remove from WebDriver.